Chore/firestore queries

BeatriceDuncan opened the pull request:
### Notion card link(s)

- [Chore: adjust firebase calls to conditionally query data by a 'wildcard' customerId if customerId isn't matched](https://www.notion.so/normative/Chore-adjust-firebase-calls-to-conditionally-query-data-by-a-wildcard-customerId-if-customerId-is-3e116aec6a354ba280aff18e4fc1ae97)

### Description (Proposed changes, additional context, chosen approach/alternative approaches, gotchas, etc)

- Adds a conditional statement to collection queries as we discussed.
- Adds optional chaining because some of the test data doesn't have all data fields (e.g. not all sensors have a systemIds array or a datachain).
- The 'wildcard' customerId is "allCustomers", this will need to be set in the Heroku config vars as the customerId for the [staging environment](https://dashboard.heroku.com/apps/dtpr-staging-normative/settings).
- I think we will still may need UI for instances where customerId isn't matched (captured in [this](https://www.notion.so/normative/Feature-UI-for-customerId-error-e178f708c814402ba9541c619f7903ba) card).

### Steps for reviewer(s) to reproduce/test

Using review app:
- In the [review app settings](https://dashboard.heroku.com/apps/dtpr-sticker-chore-fire-oo5kgr/settings) the config var for customerId had been set to "allCustomers". You should be able to open & view pages for each of these links (all have different customerIds):
[sensor 1](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/sensors/mULBhQ3KFJqUYtyWWpJT)
[place 1](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/places/bCHv97rub6Srk1uaZHMd)
[sensor 2](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/sensors/Jg9UXqD9cpM3zN31Evh0)
[place 2](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/places/xOQIMRMqwyTleqWxRJuV)

- Change the customerId config var to "XG7XJwK2bmf5PrtOJjlX" (you may need to re-deply the app) and you should only be able to open [place 1](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/places/bCHv97rub6Srk1uaZHMd) and [sensor 1](https://dtpr-sticker-chore-fire-oo5kgr.herokuapp.com/sensors/mULBhQ3KFJqUYtyWWpJT), but none of the other links above.

Using localhost:
- Change customerId in your .env to "all Customers" and test these links (all have different customerIds):
[sensor 1](http://localhost:3000/sensors/mULBhQ3KFJqUYtyWWpJT)
[place 1](http://localhost:3000/places/bCHv97rub6Srk1uaZHMd)
[sensor 2](http://localhost:3000/sensors/Jg9UXqD9cpM3zN31Evh0)
[place 2](http://localhost:3000/places/xOQIMRMqwyTleqWxRJuV)

- Change the customerId in .env to "XG7XJwK2bmf5PrtOJjlX" and only [sensor 1](http://localhost:3000/sensors/mULBhQ3KFJqUYtyWWpJT) & [place 1](http://localhost:3000/places/bCHv97rub6Srk1uaZHMd) links should work.

BeatriceDuncan committed:
Adds conditional customerId statement to data queries

BeatriceDuncan committed:
Removes comments

BeatriceDuncan committed:
Adds optional chaining to systemIds

BeatriceDuncan committed:
Adds optional chaining for datachain as some sensors don't have datachain

BeatriceDuncan committed:
Adds temporary ui on customerId error for testing purposes

tonybennett commented on src/sideEffects/firebase.ts:
```diff
@@ -39,9 +39,17 @@ export async function getPlace(
-    const placesSnapshot = await placesRef
-      .where('customerId', '==', REACT_APP_CUSTOMER_ID)
-      .get();
+    let placesSnapshot = null;
```
Minor thing but IMO `let placesSnapshot` is marginally better than setting null since you are changing the inferred types. Basically there is no need to give let a value.

tonybennett commented on src/sideEffects/firebase.ts:
```diff
@@ -39,9 +39,17 @@ export async function getPlace(
+      placesSnapshot = await placesRef
+        .where('customerId', '==', REACT_APP_CUSTOMER_ID)
+        .get();
+      if (!placesSnapshot.exists) {
```
Any reason this can't live below the conditional?  wouldn't it apply to both awaits?

tonybennett commented on src/sideEffects/firebase.ts:
```diff
@@ -67,14 +75,25 @@ export async function getPlace(
 
     const systems = {};
-    systemsSnapshot.forEach(async (systemDoc) => {
+    systemsSnapshot?.forEach(async (systemDoc) => {
```
Is the optional chaining required? You are throwing an error above if it's undefined

Again this is not critical but it changes potential for what data could look like

tonybennett reviewed (commented):
Only minor feedback on the code

BeatriceDuncan committed:
Removes null values from let variable declarations

BeatriceDuncan committed:
Moves customerId error outside of conditional

BeatriceDuncan committed:
Removes faulty customerId error handling

I can't work out what's changed/why the throw Error was working when I made the PR but now is preventing any data from loading. However the purpose of this task works: the customerId 'wildcard' conditional still works. But if the customerId doesn't match there is currently no UI to inform the user of the error, this has already been captured in a new task in notion.
